snapcraft/file_utils.py

```diff
@@ -106,18 +106,18 @@ def link_or_copy(source: str, destination: str,
     """
 
     try:
-        _link(source, destination, follow_symlinks)
+        link(source, destination, follow_symlinks)
     except OSError as e:
         if e.errno == errno.EEXIST and not os.path.isdir(destination):
             # os.link will fail if the destination already exists, so let's
             # remove it and try again.
             os.remove(destination)
             link_or_copy(source, destination, follow_symlinks)
         else:
-            _copy(source, destination, follow_symlinks)
+            copy(source, destination, follow_symlinks)
 
 
-def _link(source: str, destination: str, follow_symlinks: bool=False) -> None:
+def link(source: str, destination: str, follow_symlinks: bool=False) -> None:
```

Comment:
now that they are public it would be nice to force kwargs and add some docstrings if you don't mind

Reply:
This API needs to be compatible with copy2, so the only kwarg we can force is follow_symlinks, but I agree that it's an improvement.

Reply:
Fixed, thank you for catching the lack of docs!